R/colwise-mutate.R

```
@@ -96,9 +93,90 @@ summarize_if <- summarise_if
 #' @export
 summarize_at <- summarise_at
 
-#' @rdname summarise_all
+#' Mutate multiple columns
+#'
+#' @description
+#'
+#' These verbs are [scoped] variants of [mutate()] and [transmute()].
+#' They mutate a selection of variables.
+#'
+#' @inheritParams scoped
+#' @inheritParams summarise_all
+#' @return A data frame. By default, the newly created columns have the shortest
+#'   names needed to uniquely identify the output. To force inclusion of a name,
+#'   even when not needed, name the input (see examples for details).
+#' @seealso [The other scoped verbs][scoped], [vars()], [funs()]
+#'
+#' @section Grouping variables:
+#'
+#' If applied on a grouped tibble, these operations normally apply to
+#' only to the non-grouping variables.
+#'
+#' * `mutate_if()` and `transmute_if()` silently ignore the grouping
```

Why don't we use the same principle as for summary?

---

I don't know, I thought this was decided from a previous discussion: https://github.com/tidyverse/dplyr/issues/3473

It sort of makes sense to me. Here is how I'd frame the problem. There are implicit selections (with _if and _all variants) and explicit selections (with _at variants). When grouping variables are found in an explicit selection, we fail in all cases. When grouping variables are found in an implicit selection, we do what's intuitive and obvious.

It seems intuitive to me that grouping variables implicitly selected are ignored in a summarise operation. Somehow it's less obvious with mutate operations.

But this does suggest `mutate_if()` and `transmute_if()` should have the same behaviour than `mutate_all()` and `transmute_all()` (either a warning or silently ignore).

Do you prefer removing the warning? This would make the semantics more consistent (this consistency will be even more apparent in the docs of the next PR I'm working on).

---

I think it's clear that `summarise_if()` should only apply to non-grouping variables. It's obviously less clear for `mutate_if()` and friends because we've had a number of issues from confused users. I think that suggests we need at least a message. I think a warning might be too much, and I think suggesting `vars(-c(!!!groups(df)))` is going to confuse most people. That might be better left as advice for the docs.